File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int maxArea = 0;

        while (left < right) {
            int h = Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, h * (right - left));
            while (left < right && height[left] <= h) {
                left++;
            }
            while (left < right && height[right] <= h) {
                right--;
            }
        }

        return maxArea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        max_area = 0
        left = 0
        right = len(height)-1

        while(left < right):
            curr_height = min(height[left],height[right])
            curr_water = curr_height*(right-left)
            max_area = max(max_area,curr_water)
            
            while(left < right and height[left] <= curr_height):
                left += 1
            while(left < right and height[right] <= curr_height):
                right -= 1
        
        return max_area

        